frontend-site/src/components/Home/OfficerListing.vue

```diff
@@ -1,32 +1,65 @@
 <template>
   <v-layout row wrap class="officer-listing">
-    <v-flex
-      xs12 sm6 md4
-      v-for="(officer, i) in officers"
-      :key="i">
-      <officer-card :officer="officer"/>
+    <template v-if="!isLoading">
+      <v-flex
+        xs12 sm6 md4
+        v-for="(officer, i) in officers"
+        :key="i">
+        <officer-card :officer="officer"/>
+      </v-flex>
+      <v-flex v-if="officers.length === 0">
+        <v-card class="text-xs-center">
+          <v-card-text>No officer data found.</v-card-text>
+        </v-card>
+      </v-flex>
+    </template>
+    <v-flex v-else>
+      <v-card class="text-xs-center">
+          <v-card-text>Loading officer data...</v-card-text>
+        </v-card>
     </v-flex>
   </v-layout>
 </template>
 
 <script>
 import OfficerCard from '@/components/OfficerCard';
-import { mapState, mapActions } from 'vuex';
+import lugApi from '@/modules/LugApi';
 
 export default {
   components: {
     OfficerCard,
   },
   computed: {
-    ...mapState('officers', {
-      officers: 'data',
-    }),
-  },
-  methods: {
-    ...mapActions('officers', ['updateData']),
+    // TODO: factor out if other things need this
+    currentSemester () {
+      const currentDate = new Date();
+      const currentMonth = currentDate.getMonth() + 1; // getMonth is 0 indexed, so add 1
+      const currentYear = currentDate.getUTCFullYear();
+      let currentSemester = '';
+      if (currentMonth <= 5) { // up to May
+        currentSemester = 'SPRING';
+      } else if (currentMonth <= 8) { // up to August
+        currentSemester = 'SUMMER';
+      } else {
+        currentSemester = 'FALL';
+      }
+      return `${currentSemester}_${currentYear}`;
+    },
   },
+  data: () => ({
+    officers: [],
+    isLoading: true,
+  }),
   async mounted () {
-    await this.updateData();
+    try {
+      this.officers = await lugApi.getOfficers({
+        semester: this.currentSemester,
+      });
+    } catch (err) {
+      console.error(err);
+    } finally {
+      this.isLoading = false;
+    }
```

Comment:
It seems redundant to wrap a promise in a try-catch. Does `getOfficers.catch().finally()` work?

Reply:
I'm using the async/await syntax here, which supports using try/catch/finally for Promise-based objects. It could be done with something like below if I were using promises directly.
```js
lugApi.getOfficers(/* params here */)
  .then(officers => this.officers = officers)
  .catch(err => console.error(err))
  .finally(() => this.isLoading = false);
```

Reply:
When I was messing around with it, I seem to remember you could mix async/await and chained promise calls. That's probably not idiomatic though. This is good as is.

Reply:
You can mix them. If I were to do it here, it'd probably be something like below. Though, I'd only mix async/await and chained promises for any data transformations (e.g. if the officers array were buried in a response object, I would chain it here).
```js
this.officers = await lugApi.getOfficers(/* params here */)
  .catch(err => { console.error(err); return [] })
  .finally(() => this.isLoading = false);
```